File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        seen = {}

        for i in range(0,len(nums)):
            if nums[i]  in seen:
                seen[nums[i]]+=1
            else:
                seen[nums[i]]=1

        for i in seen:
            if seen[i] > (len(nums)/2):
                majority = i
                
        
        return majority

        

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {

        HashMap <Integer,Integer> frequency = new HashMap<>();

        for(int i = 0;i<nums.length;i++){
            if(frequency.containsKey(nums[i])){
                frequency.put(nums[i],(frequency.get(nums[i]))+1);
            }else{
                frequency.put(nums[i],1);
            }
        }

        for(int i : nums){
            if(frequency.get(i)>(nums.length/2)){
                return i;
            }
        }

        return -1;
        
    }
}